odoo/addons/test_new_api/tests/test_one2many.py

```diff
@@ -278,3 +278,22 @@ def test_reward_line_delete(self):
             order.write({
                 'line_ids': [(4, line0.id), (4, line1.id)],
             })
+
+    def test_child_of(self):
+        Team = self.env['test_new_api.team']
+        Member = self.env['test_new_api.team.member']
+
+        team1 = Team.create({'name': 'ORM'})
+        team2 = Team.create({'name': 'Bugfix', 'parent_id': team1.id})
+        team3 = Team.create({'name': 'Support', 'parent_id': team2.id})
+
+        Member.create({'name': 'Raphael', 'team_id': team1.id})
+        member2 = Member.create({'name': 'Noura', 'team_id': team3.id})
+        Member.create({'name': 'Ivan', 'team_id': team2.id})
+
+        # In this specific case...
+        self.assertEqual(member2.id, member2.team_id.parent_id.id)
+
+        # ...we had an infinite recursion on making the following search.
+        with self.assertRaises(ValueError):
+            Team.search([('member_ids', 'child_of', team2.id)])
```

Comment:
Doesn't this make sense though?
Shouldn't this return all the teams that have members that are in `team2` or one of it's subteams? So in `team2` and/or `team3`, so both since they both have a least one member.

Reply:
Consider this example instead:

            Team.search([('member_ids', 'child_of', 9)])

`9` should be an id of some member, right?

Reply:
Not if you decided that the parents of members are teams, like you declared it in your fields.
Otherwise you should for big to have a `_parent_field` that is not related to the same model, but I'm fairly certain we don't want that